lib/archive/unpack_unix.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package archive
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"errors"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	// fileWriteFlags is a collection of flags configuring our writes for general tar behavior
+	//
+	// O_CREATE = Create file if it does not exist
+	// O_TRUNC = truncate file to 0 length if it does exist(overwrite the file)
+	// O_WRONLY = We use this since we do not intend to read, we only need to write.
+	fileWriteFlags = os.O_CREATE | os.O_TRUNC | os.O_WRONLY
+)
+
+// unpack will unpack the given tarstream(if it is a tar stream) on the local filesystem based on the specified root
+// combined with any rebase from the path spec
+//
+// the pathSpec will include the following elements
+// - include : any tar entry that has a path below(after stripping) the include path will be written
+// - strip : The strip string will indicate the
+// - exlude : marks paths that are to be excluded from the write operation
+// - rebase : marks the the write path that will be tacked onto (appended or prepended? TODO improve this comment) the "root". e.g /tmp/unpack + /my/target/path = /tmp/unpack/my/target/path
+func InvokeUnpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec) error {
```

Comment:
it looks like this whole file is new, because I moved it to unpack_unix.go re: what I was talking about on the phone, but the changes are only:
Unpack renamed to InvokeUnpack
new "Unpack" all new code
a few routines in InvokeUnpack are commented out